File: src/02.string/04.transform/10.443.go
package transform

import "strconv"

func Compress(chars []byte) int {
	n := len(chars)
	k := 0

	for i, start := 0, 0; i < n; start = i {
		for i < n && chars[i] == chars[start] {
			i++
		}

		chars[k] = chars[start]
		k++
		count := i - start
		if count > 1 {
			countStr := strconv.Itoa(count)
			for j := 0; j < len(countStr); j++ {
				chars[k] = countStr[j]
				k++
			}
		}
	}

	return k
}

File: src/02.string/04.transform/10.443.ts
export function compress(chars: string[]): number {
    const n = chars.length;
    let k = 0;
    for (let i = 0, start = 0; i < n; start = i) {
        while (i < n && chars[i] === chars[start]) {
            i++;
        }
        chars[k++] = chars[start];
        const count = i - start;
        if (count > 1) {
            const countStr = "" + count;
            for (let j = 0; j < countStr.length; j++) {
                chars[k++] = countStr[j];
            }
        }
    }
    chars.length = k;
    return k;
}
